### PR TITLE
feat(deploy): Copy any custom user profiles

### PR DESCRIPTION
**Pull request by lwander**
The goal of this is to enable supplying custom rosco packer templates & scripts more easily.

### CODE REVIEW

**Comment by lwander** [comment]
https://github.com/spinnaker/halyard/issues/409

File: halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/GenerateService.java [hunk 1]
```diff
@@ -123,45 +124,57 @@ public ResolvedConfiguration generateConfig(String deploymentName, List<String>
...
+    if (!relativePath.isEmpty()) {
+      filePrefix = relativePath + File.separator;
+    } else {
+      filePrefix = relativePath;
+    }
+
+    File currentPath = new File(basePath, relativePath);
+    return Arrays.stream(currentPath.listFiles())
+        .map(f -> f.isFile() ? Collections.singletonList(filePrefix + f.getName()) : aggregateProfilesInPath(basePath, filePrefix + f.getName()))
+        .reduce(new ArrayList<>(), (a, b) -> {
```
Comment (jtk54) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/GenerateService.java]: Can you name these more clearly?

Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/GenerateService.java]: `a, b`? this is just a standard reducer, I don't know what else to name them... `left, right`? `one, two`?

Comment (jtk54) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/GenerateService.java]: What do those represent?

Comment (lwander) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/GenerateService.java]: Lists being joined

Comment (jtk54) [inline halyard-deploy/src/main/java/com/netflix/spinnaker/halyard/deploy/services/v1/GenerateService.java]: Ok this is fine how it is, just dense on the first pass.

**Pull request merged**